assayist/processor/container_go_analyzer.py

```diff
@@ -105,21 +96,100 @@ def run(self):
+        :param str srcdir: path to source code to examine
+        """
+        dist_git_excludes = [
+            # Files expected to be local and not upstream.
+            '.git*',
+            'Makefile',
+            'sources',
+            'Dockerfile*',
+            'additional-tags',
+            'content_sets.yml',
+            'container.yaml',
+            '.oit',
+            'public',
+            'scripts',
+        ]
+
+        # In case we need to provide an import path, see if there is
+        # an io.openshift.source-repo-url label in the Dockerfile to help
+        # us guess one.
+        import_path_from_source_url = self._get_import_path_override(srcdir)
+
+        # If we have a reasonable guess, see if we will need to use it.
+        import_path = []
+        if (import_path_from_source_url and
+                not self._import_paths_known(srcdir, excludes=dist_git_excludes)):
+            import_path = ['-importpath', import_path_from_source_url]
+
+        self._process_source_code(source_location, srcdir,
+                                  import_path=import_path,
+                                  excludes=dist_git_excludes)
+
+    def _get_import_path_override(self, srcdir):
+        """Look inside the Dockerfile for a named label.
+
+        :param srcdir: path to source code to examine
+        :return: import path override, or None
+        :rtype str/None:
+        """
+        label = 'io.openshift.source-repo-url'
+        try:
+            df = DockerfileParser(srcdir, cache_content=True)
+        except IOError:
+            log.exception('Unable to read Dockerfile')
+            return None
+
+        try:
+            repo = df.labels[label]
+        except KeyError:
+            log.debug(f'No {label} label in Dockerfile')
+            return None
+        except:  # noqa:E722
+            log.exception('Failed to process Dockerfile; ignoring')
+            return None
+
+        # Convert it to an import path by stripping off the scheme.
+        (_, _, import_path) = repo.rpartition('://')
+        if not import_path:
+            return None
+
+        return import_path
+
+    def _run_backvendor(self, srcdir, import_path=None, excludes=None,
+                        opts=None):
+        """Run backvendor and returns its output.
+
+        :param srcdir: path to source code to examine
+        :param str/None import_path: import path for top-level module
+        :param list/None excludes: list of globs to ignore
+        :param list/None opts: any additional parameters
+        :return: output from command
+        :rtype: (str, str)
         """
-        # Run backvendor.
         with tempfile.NamedTemporaryFile(mode='wt') as excludes_file:
-            options = ['-debug', '-template', self.BACKVENDOR_TEMPLATE]
+            options = ['-debug', '-x', '-template', self.BACKVENDOR_TEMPLATE]
+            if import_path:
+                options += ['-importpath', import_path]
```

Comment:
Parameter `import_path` is a list (line 130) and then `cmd` will contain list within list which is not allowed python

```
Running ContainerGoAnalyzer...
INFO:assayist_processor:Running ['backvendor', '-debug', '-x', '-template', '\t{{.Ver}}\t{{.Repo}}\t{{.Rev}}', '-importpath', ['.git*', 'Makefile', 'sources', 'Dockerfile*', 'additional-tags', 'content_sets.yml', 'container.yaml', '.oit', 'public', 'scripts'], '-only-importpath', '/home/jenkins/workspace/assayist-pipeline-test/playground/source']

Traceback (most recent call last):
  File "/src/scripts/run-analyzers.py", line 30, in <module>
    analyzer(input_dir).main()
  File "/src/assayist/processor/base.py", line 109, in main
    self.run()
  File "/src/assayist/processor/container_go_analyzer.py", line 72, in run
    self._process_git_source(source_location, srcdir)
  File "/src/assayist/processor/container_go_analyzer.py", line 122, in _process_git_source
    not self._import_paths_known(srcdir, excludes=dist_git_excludes)):
  File "/src/assayist/processor/container_go_analyzer.py", line 206, in _import_paths_known
    self._run_backvendor(srcdir, excludes, opts=['-only-importpath'])
  File "/src/assayist/processor/container_go_analyzer.py", line 187, in _run_backvendor
    stderr=subprocess.PIPE)
  File "/usr/lib64/python3.6/subprocess.py", line 709, in __init__
    restore_signals, start_new_session)
  File "/usr/lib64/python3.6/subprocess.py", line 1275, in _execute_child
    restore_signals, start_new_session, preexec_fn)
TypeError: expected str, bytes or os.PathLike object, not list
```

Reply:
Thanks, fixed.